src/app/resources/iis_scripts/admin_api_util.ps1

```diff
@@ -0,0 +1,120 @@
+param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('install', 'permission-check')]
+    [string]
+    $command,
+
+    [string]
+    $downloadFrom,
+
+    ## Not doing anything with this parameter yet
+    [string]
+    $sessionId,
+
+    [string]
+    $serviceName = "Microsoft IIS Administration",
+
+    [string]
+    $iisAdminGroup = "WAC-IISAdmin"
+)
+
+$ErrorActionPreference = "Stop"
+
+######################## Param checks ###############################
+$install = $command -eq 'install'
+
+if ($install -and $(!$downloadFrom)) {
+    throw "Missing download location for install command"
+}
+
+######################## Utilities ###############################
+function InstallAPI {
+    $installer = Join-Path $env:TEMP iis-administration-setup.exe
+    Invoke-WebRequest $downloadFrom -OutFile $installer
+    & $installer /s
+
+    $waitPeriod = 1
+    $remainingTime = 600
+
+    while (!(Get-Service $serviceName -ErrorAction SilentlyContinue | Where-Object {$_.Status -eq "Running"})) {
+        Start-Sleep -Seconds $waitPeriod
+        $remainingTime -= $waitPeriod
+        if ($remainingTime -lt 0) {
+            throw "timeout waiting for service to start"
+        }
+    }
+}
+
+function EnsureAdminGroup($user) {
+    New-LocalGroup -Name $iisAdminGroup -ErrorAction SilentlyContinue | Out-Null

```

Comment:
What if the group already exists?

Reply:
That's why I have SilentlyContinue set to make the code shorter.

Reply:
I actually change the code to check for group existence and membership before creating or modifying. Because on second thought it would be more readable